src/thor/triplegbuilder.cc

```diff
@@ -813,41 +889,37 @@ TripLeg_Edge* AddTripEdge(const AttributesController& controller,
 
   // Index of the directed edge within the tile
   uint32_t idx = edge.id();
-
   TripLeg_Edge* trip_edge = trip_node->mutable_edge();
 
   // Get the edgeinfo
   auto edgeinfo = graphtile->edgeinfo(directededge);
 
   // Add names to edge if requested
   if (controller(kEdgeNames)) {
-    auto names_and_types = edgeinfo.GetNamesAndTypes(true);
+    const auto names_and_types = edgeinfo.GetNamesAndTypes(true);
     trip_edge->mutable_name()->Reserve(names_and_types.size());
-    std::unordered_map<uint8_t, std::pair<uint8_t, std::string>> pronunciations =
-        edgeinfo.GetPronunciationsMap();
+    const auto linguistics = edgeinfo.GetLinguisticMap();
+
     uint8_t name_index = 0;
     for (const auto& name_and_type : names_and_types) {
-      if (std::get<2>(name_and_type) != 0) {
-        // Skip the tagged names
-        continue;
-      }
-
-      auto* trip_edge_name = trip_edge->mutable_name()->Add();
-      // Assign name and type
-      trip_edge_name->set_value(std::get<0>(name_and_type));
-      trip_edge_name->set_is_route_number(std::get<1>(name_and_type));
-      std::unordered_map<uint8_t, std::pair<uint8_t, std::string>>::const_iterator iter =
-          pronunciations.find(name_index);
-
-      // Assign pronunciation alphabet and value if one exists
-      if (iter != pronunciations.end()) {
-        auto* pronunciation = trip_edge_name->mutable_pronunciation();
-        pronunciation->set_alphabet(GetTripPronunciationAlphabet(
-            static_cast<valhalla::baldr::PronunciationAlphabet>((iter->second).first)));
-        pronunciation->set_value((iter->second).second);
+      switch (std::get<2>(name_and_type)) {
+        case kNotTagged: {
+          ProcessNonTaggedValue(trip_edge->mutable_name()->Add(), linguistics, name_and_type,
+                                name_index);
+          break;
+        }
+        case kTunnelTag: {
```

Comment:
why does only tunnel ever get used, i wonder why bridge is left out

Reply:
The mothership did not care about them yet.  Updating